src/plugins/data/server/index_patterns/fetcher/lib/field_capabilities/field_capabilities.ts

```diff
@@ -69,7 +70,8 @@ export async function getFieldCapabilities(
         readFromDocValues: false,
       })
     )
-    .map(mergeOverrides);
+    .map(mergeOverrides)
+    .filter((field) => (filters?.aggregatable ? field.aggregatable === true : true));
```

Comment:
If this is a particularly hot code path, it might be worth making this `.filter` conditional rather than always adding an extra loop.